src/core/Table/TableRowMemoized.tsx

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { Row, TableRowProps, TableState } from 'react-table';
+import { useIntersection } from '../utils/hooks/useIntersection';
+import { getCellStyle } from './utils';
+
+const TableRow = <T extends Record<string, unknown>>(props: {
+  row: Row<T>;
+  rowProps: TableRowProps;
+  isLast: boolean;
+  onRowInView: React.MutableRefObject<((rowData: T) => void) | undefined>;
+  onBottomReached: React.MutableRefObject<(() => void) | undefined>;
+  intersectionMargin: number;
+  state: TableState<T>; // Needed for explicitly checking selection changes
+}) => {
+  const {
+    row,
+    rowProps,
+    isLast,
+    onRowInView,
+    onBottomReached,
+    intersectionMargin,
+  } = props;
+
+  const onIntersect = React.useCallback(() => {
+    onRowInView.current?.(row.original);
+    isLast && onBottomReached.current?.();
+  }, [isLast, onBottomReached, onRowInView, row.original]);
+
+  const setRef = useIntersection(onIntersect, {
```

Comment:
What does it mean "setRef" ? Dpesnt it just add observer..?

Reply:
It is used in `ref` prop so for me it makes sense.

Reply:
Well, `setRef` implies that it's a setter (like a `setState` call). If it's used directly in `ref` prop, it should just be called `rowRef` or something.

Reply:
Leaving `setRef` in useIntersection.tsx? Guess it could be fine since it's a `useCallback`